paper-server/patches/sources/net/minecraft/world/level/ServerExplosion.java.patch

```diff
@@ -74,8 +74,13 @@
 +            for (Entity entity : list) { // Paper - used in loop
                  if (!entity.ignoreExplosion(this)) {
                      double d = Math.sqrt(entity.distanceToSqr(this.center)) / f;
++                    // Paper start - avoid infinite damage/velocity when distance and radius are 0
++                    if (Double.isNaN(d)) {
++                        d = Double.POSITIVE_INFINITY;
++                    }
++                    // Paper end
```

Comment:
Probably just make this a single line so you don't need start / end comments but just a single one.
Also, include an explanation for +inf here (e.g. that 0+h / 0 = +inf).